### PR TITLE
CHAT-832 raise clear errors when trying to build tokens without secret or on browser

### PR DESCRIPTION
**Pull request by tbarbugli**

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -374,6 +374,9 @@ export class StreamChat {
 	 * @return {string} Returns a token
 	 */
 	createToken(userID, exp) {
+		if (this.secret == null) {
```
Comment (ferhatelmas) [inline src/client.js]: isn't it better to check falsy values?

File: src/signing.js [hunk 2]
```diff
@@ -22,6 +22,13 @@ export function JWTUserToken(apiSecret, userId, extraData = {}, jwtOptions = {})
 		...extraData,
 	};
 
+	// make sure we return a clear error when jwt is shimmed (ie. browser build)
+	if (jwt == null || jwt.sign == null) {
+		throw Error(
+			`Unable to find jwt crypto, if you are getting this error is probably because you are trying to generate tokens on browser or React Native (or other environment where crypto functions are not available). Please Note: token should only be generate server-side.`,
```
Comment (ferhatelmas) [inline src/signing.js]: would be nice if wrapped.

Comment (ferhatelmas) [inline src/signing.js]:
```diff
- should only be generate
+ should only be generated
```

Comment (ferhatelmas) [reply src/signing.js]: `if you are getting this error is` could be dropped, IMHO

File: src/client.js (hunk 1 above)
Comment (tbarbugli) [reply src/client.js]: not needed in this case (you can see that in the client constructor)

**Pull request merged**